src/v/redpanda/admin_server.cc

```diff
@@ -2153,6 +2159,28 @@ void admin_server::register_features_routes() {
       });
 }
 
+ss::future<std::optional<cluster::drain_manager::drain_status>>
+get_node_drain_status(
+  cluster::controller* controller,
+  model::node_id id,
+  model::timeout_clock::time_point deadline) {
+    if (controller == nullptr) {
+        throw ss::httpd::server_error_exception(
+          "Unexpected error: controller undefined");
```

Comment:
I think you should log an error message here and return a generic error to the client like `internal server error encountered.`